### ISSUE TITLE
Can't override static builder for the builder-in-disguise pattern

### ISSUE THREAD

**Issue by nezda** [immutables/immutables#1467]
Can't override static builder for the builder-in-disguise pattern (from https://github.com/immutables/immutables/issues/234#issuecomment-170574928) which is referenced in the docs

> [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.11.0:compile (default-compile) on project value-fixture: Compilation failure
[ERROR] /Users/nezda/code/immutables/value-fixture/target/generated-sources/annotations/org/immutables/fixture/modifiable/ImmutableBuilderInDisguise.java:[156,51] error: builder() in ImmutableBuilderInDisguise cannot hide builder() in BuilderInDisguise
[ERROR]   return type Builder is not compatible with BuilderInDisguiseBuilder

see https://github.com/immutables/immutables/pull/1466

**Comment by elucash** [comment]
I may not understand it well, I'll reread tomorrow. But from the first impression.
One static method cannot hide another if return type differs (iirc) [given that same signature - no params]. In order to un-clash those signatures something have to be done.

* Rename default builder method  (`@Style.builder = "anotherBuilder"  |  "new" `)
* Disable standard builder (`@Value.Immutable(builder = false`), might require `@Style.allParameters`
* or smth like that...

**Comment by nezda** [comment]
You've got it exactly right @elucash . I tried adding `@Value.Immutable(builder = false)` but as you said, probably requires `@Style.allParameters` as it failed to compile with error below. If I'm interpreting the builder-in-disguise recipe correctly, this is sort of a bug / unanticipated? Not being able to use the name `builder` hobbles the disguise a little bit depending on the name you pick (for my actual use case I called it `modifiableBuilder` in the abstract class. Thinking about it more, renaming the immutable builder with something like `@Style.builder = "immutableBuilder"` seems better and works fine. Maybe the documentation for this feature should suggest this?

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.11.0:compile (default-compile) on project value-fixture: Compilation failure
[ERROR] /Users/nezda/code/immutables/value-fixture/src/org/immutables/fixture/modifiable/BuilderInDisguise.java:[19,25] error: Attribute 'name' is mandatory and should be a constructor @Value.Parameter when builder is disabled or there are other constructor parameters
```

**Comment by elucash** [comment]
Yes, I would say the clash is a natural and unanticipated consequence. I dunno, maybe adding some validation or documentation might help, but just not enough time/resources/motivation to polish these rarely used areas. These renamings, as they are, exist as a sharp tool. If I [ever] will rewrite immutables I would, probably, construct and validate generation model and validate it thoroughly (extensibility -> plugins -> need to detect / resolve conflicts). Any PRs for javadocs, or if we just copy paste some comments into builder-in-disguise ticket – always welcome
